frontend/src/utils/utils.ts

```diff
@@ -753,3 +753,21 @@ export function setHeader(init: RequestInit, name: string, value: string) {
         (init.headers as Record<string, string>)[name] = value;
     }
 }
+
+// very simple retrier utility for allowing some retries, if we ended up using it more often we should consider making it more elaborate
+
+export const retrier = (
+    operation: any, { attempts = Infinity, delayTime = 100 }) => new Promise((resolve, reject) => {
+        return operation()
+            .then(resolve)
+            .catch((reason: unknown) => {
+                if (attempts > 0) {
+                    return delay(delayTime)
+                        .then(retrier.bind(null, operation, { attempts: attempts - 1, delayTime }))
```

Comment:
If `attempts` are `Infinity` this part should be redundant, you could do
```suggestion
                        .then(retrier.bind(null, operation, { attempts: Number.isFinite(attempts) ? attempts - 1 : attempts, delayTime }))
```

Reply:
I don't see the value of making this more illegible when the performance impact is basically 0

Reply:
yea, leave it as is.